server/src/main/java/org/apache/calcite/sql/ddl/SqlColumnDeclaration.java

```diff
@@ -70,17 +71,14 @@
     if (dataType.getNullable() != null && !dataType.getNullable()) {
       writer.keyword("NOT NULL");
     }
-    if (dataType.getUppercase() != null) {
-      String keyword = dataType.getUppercase()
-          ? "UPPERCASE"
-          : "NOT UPPERCASE";
-      writer.keyword(keyword);
-    }
-    if (dataType.getCaseSpecific() != null) {
-      String keyword = dataType.getCaseSpecific()
-          ? "CASESPECIFIC"
-          : "NOT CASESPECIFIC";
-      writer.keyword(keyword);
+    List<SqlColumnAttribute> columnAttributes = dataType.getColumnAttributes();
+    if (columnAttributes != null && columnAttributes.size() > 0) {
+      SqlWriter.Frame frame = writer.startList("", "");
```

Comment:
nit: is this needed since the starting and ending tokens are both empty strings?